src/Components/WebAssembly/WebAssembly/src/Http/WebAssemblyHttpRequestMessageExtensions.cs

```diff
@@ -0,0 +1,164 @@
+                _ => throw new InvalidOperationException($"Unsupported enum value {requestMode}.")
+            };
+
+            return SetFetchOption(requestMessage, "mode", stringOption);
+        }
+
+        /// <summary>
+        /// Configures a value for the 'integrity' option for the HTTP request.
+        /// </summary>
+        /// <param name="requestMessage">The <see cref="HttpRequestMessage"/>.</param>
+        /// <param name="integrity">The subresource integrity descriptor.</param>
+        /// <returns>The <see cref="HttpRequestMessage"/>.</returns>
+        public static HttpRequestMessage SetIntegrity(this HttpRequestMessage requestMessage, string integrity)
+            => SetFetchOption(requestMessage, "integrity", integrity);
+
+        /// <summary>
+        /// Configures a value for the HTTP request.
+        /// </summary>
+        /// <param name="requestMessage">The <see cref="HttpRequestMessage"/>.</param>
+        /// <param name="name">The name of the HTTP fetch option.</param>
+        /// <param name="value">The value, which must be JSON-serializable.</param>
+        /// <returns>The <see cref="HttpRequestMessage"/>.</returns>
+        /// <remarks>
+        /// See https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch
+        /// </remarks>
+        public static HttpRequestMessage SetFetchOption(this HttpRequestMessage requestMessage, string name, object value)
+        {
+            if (requestMessage is null)
+            {
+                throw new ArgumentNullException(nameof(requestMessage));
+            }
+
+            const string FetchRequestOptionsKey = "FetchRequestOptions";
+            IDictionary<string, object> fetchOptions;
+
+            if (requestMessage.Properties.TryGetValue(FetchRequestOptionsKey, out var entry))
+            {
+                fetchOptions = (IDictionary<string, object>)entry;
+            }
+            else
+            {
+                fetchOptions = new Dictionary<string, object>();
+                requestMessage.Properties[FetchRequestOptionsKey] = fetchOptions;
+            }
+
+            fetchOptions[name] = value;
+
+            return requestMessage;
+        }
+
+        /// <summary>
+        /// Configures streaming response for the HTTP request.
+        /// </summary>
+        /// <param name="requestMessage">The <see cref="HttpRequestMessage"/>.</param>
+        /// <param name="streamingEnabled"><see langword="true"> if streaming is enabled; otherwise false.</param>
+        /// <returns>The <see cref="HttpRequestMessage"/>.</returns>
+        /// <remarks>
+        /// This API is only effective when the browser HTTP Fetch supports streaming.
+        /// See https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream.
+        /// </remarks>
+        public static HttpRequestMessage SetStreamingEnabled(this HttpRequestMessage requestMessage, bool streamingEnabled)
```

Comment:
@JamesNK

Reply:
This a wrapper over the existing WASM HttpHandler right? Is that existing handler using XHR or Fetch under the hood? I'm guessing by the XML doc here that it is using Fetch. I'm mostly curious because the gRPC-Web JavaScript library uses XHR. I don't think there is a reason one has to be used for gRPC-Web to work.

Is it worth mentioning in the comment that you have to use the async methods on the stream? The reason streaming is disable at the moment is sync reads from third-party libraries like CsvHelper break.

Will you make the change in the underlying handler to read the property value, or will the Mono team?

Another thing to think about is what happens if the default value changes. Currently streaming is disabled but there might be a point in time when the ecosystem is using the Stream async and the default changes.

Reply:
Here is the corresponding PR on the mono side: https://github.com/mono/mono/pull/19054